### PR TITLE
feat: Implement useUncontrolledFocus hook

### PR DESCRIPTION
**Pull request by ling1726**
Exposes a hook to enable uncontrolled tabster to improve interop between v8 FocusZone and FocusTrapZone.

Also updates `useTabster` to use the new external focus trap check so that it works with v8 FocusTrapZone

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">67</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/Avatar-mount_base.html">611</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/Avatar-mount.html">661</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/Button-mount_base.html">310</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/Button-mount.html">327</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/Field-mount_base.html">1132</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/Field-mount.html">1136</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProvider-mount_base.html">713</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProvider-mount.html">731</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-mount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-mount.html">81</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">67</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">71</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/MakeStyles-mount_base.html">842</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/MakeStyles-mount.html">860</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/Persona-mount_base.html">1761</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/Persona-mount.html">1674</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/SpinButton-mount_base.html">1342</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29895/perf-test-react-components/SpinButton-mount.html">1385</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                   |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                  Change |
| :---------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr> |   `93.44 kB`<br />`28.513 kB` |  `93.591 kB`<br />`28.552 kB` | `151 B` <br />`39 B`  |
| <samp>react-calendar-compat</samp> <br /> <abbr title='bundle-size/Calendar.fixture.js'>Calendar Compat</abbr>                      | `141.822 kB`<br />`36.959 kB` | `141.973 kB`<br />`36.998 kB` | `151 B` <br />`39 B`  |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                      |  `93.423 kB`<br />`27.059 kB` |  `93.574 kB`<br />`27.101 kB` | `151 B` <br />`42 B`  |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                |  `88.247 kB`<br />`25.559 kB` |  `88.398 kB`<br />`25.603 kB` | `151 B` <br />`44 B`  |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr>                | `213.382 kB`<br />`59.599 kB` | `213.533 kB`<br />`59.648 kB` | `151 B` <br />`49 B`  |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>          |  `95.021 kB`<br />`28.381 kB` |  `95.172 kB`<br />`28.425 kB` | `151 B` <br />`44 B`  |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoButton.fixture.js'>InfoButton</abbr>                              | `131.616 kB`<br />`41.398 kB` | `131.767 kB`<br />`41.438 kB` | `151 B` <br />`40 B`  |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoLabel.fixture.js'>InfoLabel</abbr>                                | `135.302 kB`<br />`42.548 kB` | `135.453 kB`<br />`42.586 kB` | `151 B` <br />`38 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                | `140.971 kB`<br />`43.239 kB` | `141.122 kB`<br />`43.283 kB` | `151 B` <br />`44 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>   | `143.657 kB`<br />`43.747 kB` | `143.808 kB`<br />`43.788 kB` | `151 B` <br />`41 B`  |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                       | `120.349 kB`<br />`37.903 kB` |   `120.5 kB`<br />`37.958 kB` | `151 B` <br />`55 B`  |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/TagGroup.fixture.js'>TagGroup</abbr>                                        |  `74.271 kB`<br />`22.229 kB` |  `74.422 kB`<br />`22.269 kB` | `151 B` <br />`40 B`  |
| <samp>react-toast</samp> <br /> <abbr title='bundle-size/Toast.fixture.js'>Toast (including Toaster)</abbr>                         |  `93.195 kB`<br />`27.948 kB` |  `93.346 kB`<br />`27.986 kB` | `151 B` <br />`38 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                       | `84.293 kB`<br />`23.456 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                    | `50.534 kB`<br />`15.923 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                          |  `20.064 kB`<br />`7.894 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                  |  `65.19 kB`<br />`20.259 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                    | `40.074 kB`<br />`11.292 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                    |  `47.43 kB`<br />`12.787 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                            | `44.848 kB`<br />`12.668 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                          | `52.862 kB`<br />`14.254 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                        | `57.122 kB`<br />`13.194 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                              |  `13.423 kB`<br />`5.481 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                              |  `15.666 kB`<br />`6.235 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                            |  `14.374 kB`<br />`5.854 kB` |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr>                              | `34.424 kB`<br />`11.829 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `92.437 kB`<br />`30.132 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `90.847 kB`<br />`29.792 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                         |    `17.4 kB`<br />`6.977 kB` |
| <samp>react-message-bar</samp> <br /> <abbr title='bundle-size/Default.fixture.js'>MessageBar (all components)</abbr>         |   `25.54 kB`<br />`9.102 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                 | `57.425 kB`<br />`17.798 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                    |  `12.571 kB`<br />`4.594 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                  |   `23.008 kB`<br />`8.48 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                       | `31.437 kB`<br />`10.314 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                             |   `15.704 kB`<br />`6.35 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                    |  `38.52 kB`<br />`12.646 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                    | `33.902 kB`<br />`10.944 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/InteractionTag.fixture.js'>InteractionTag</abbr>                      |  `15.618 kB`<br />`6.152 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/tag.fixture.js'>Tag</abbr>                                            |  `30.341 kB`<br />`9.505 kB` |
| <samp>react-timepicker-compat-preview</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr>         |  `93.834 kB`<br />`31.42 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                 | `53.173 kB`<br />`18.712 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d82568581b73996acbfdf78966e55572e0f869f6'>d82568581b73996acbfdf78966e55572e0f869f6</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: change/@fluentui-react-components-22466e0d-b22b-4ce2-9dc7-624450174651.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "feat: export useTabsterUncontrolled hook",
```
Comment (ling1726) [inline change/@fluentui-react-components-22466e0d-b22b-4ce2-9dc7-624450174651.json:3]:
```suggestion
  "comment": "feat: export useUncontrolledFocus hook",
```

File: change/@fluentui-react-tabster-aacd5d7c-2260-4855-ace5-a3aeeb6c862b.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "feat: implement useTabsterUncontrolled hook",
```
Comment (ling1726) [inline change/@fluentui-react-tabster-aacd5d7c-2260-4855-ace5-a3aeeb6c862b.json:3]:
```suggestion
  "comment": "feat: implement useUncontrolledFocus hook",
```

**Pull request merged**